apps/ex_wire/lib/ex_wire/packet/get_snapshot_manifest.ex

```diff
@@ -0,0 +1,60 @@
+defmodule ExWire.Packet.GetSnapshotManifest do
+  @moduledoc """
+  Request a snapshot manifest in RLP form from a peer.
+
+  ```
+  `GetSnapshotManifest` [`0x11`]
+  ```
+  """
+
+  @behaviour ExWire.Packet
+
+  @type t :: %__MODULE__{}
+
+  defstruct []
+
+  @doc """
+  Given a GetSnapshotManifest packet, serializes for transport over Eth Wire Protocol.
+
+  ## Examples
+
+      iex> %ExWire.Packet.GetSnapshotManifest{}
+      ...> |> ExWire.Packet.GetSnapshotManifest.serialize()
+      []
+  """
+  @spec serialize(t) :: ExRLP.t()
+  def serialize(_packet = %__MODULE__{}) do
```

Comment:
It's just pattern matching against the packet being passed in is a struct of this module. Thus if I called: `ExWire.Packet.GetSnapshotManifest.serialize(%{})` it would fail with a type match error. Acts as a guard.